code/controllers/subsystem/changelog.dm

```diff
@@ -0,0 +1,220 @@
+/*
+	Subsystem core for ParadiseSS13 changelogs
+	Author: AffectedArc07
+
+	Basically this SS extracts changelogs from the past 30 days from the database, and cleanly formats them into HTML that the players can see
+	It only runs the extraction on initialize to ensure that the changelog doesnt change mid round, and to reduce the amount of DB calls that need to be done
+	The changelog entries are generated from the PHP scripts in tools/githubChangelogProcessor.php
+
+	This SS also handles the checking of player CL dates and informing them if it has changed
+
+*/
+
+SUBSYSTEM_DEF(changelog)
+	name = "Changelog"
+	flags = SS_NO_FIRE
+	var/current_cl_timestamp = "0" // Timestamp is seconds since UNIX epoch (1st January 1970)
+	var/ss_ready = 0 // Is the SS ready? We dont want to run procs if we have not generated yet
+	var/list/startup_clients_button = list() // Clients who connected before initialization who need their button color updating
+	var/list/startup_clients_open = list() // Clients who connected before initialization who need the CL opening
+	var/changelogHTML = "" // HTML that the changelog will use to display
+
+/datum/controller/subsystem/changelog/Initialize()
+	// This entire subsystem relies on SQL being here.
+	if(!config.sql_enabled)
+		return ..() 
+	
+	var/DBQuery/latest_cl_date = dbcon.NewQuery("SELECT UNIX_TIMESTAMP(date_merged) AS ut FROM [format_table_name("changelog")] ORDER BY date_merged DESC LIMIT 1")
+	if(!latest_cl_date.Execute())
+		var/err = latest_cl_date.ErrorMsg()
+		log_game("SQL ERROR during SSchangelog initialization L24. Error: \[[err]\]\n")
+		message_admins("SQL ERROR during SSchangelog initialization L24. Error: \[[err]\]\n")
+		// Abort if we cant do this
+		return ..()
+
+	while(latest_cl_date.NextRow())
+		current_cl_timestamp = latest_cl_date.item[1]
+
+	if(!GenerateChangelogHTML()) // if this failed to generate
+		to_chat(world, "<span class='alert'>WARNING: Changelog failed to generate. Please inform a coder/server dev</span>")
+		return ..()
+
+	ss_ready = 1
+	// Now we can alert anyone who wanted to check the changelog
+	for(var/client/C in startup_clients_button)
```

Comment:
if you're sure the list only contains clients then you can loop through it using `var/thingy` and then cast it in the loop scope. Bit faster

Reply:
The list *should* only ever contain clients so ill do that